documentation/docs/install-pmm/install-pmm-client/connect-database/mongodb.md

```diff
@@ -58,266 +64,245 @@ Run the example codes below in a `mongo` session to:
+=== "MongoDB < 8.0"
 
-    !!! caution alert alert-warning "Caution"
-        Smaller values improve accuracy but can adversly affect the performance of your server.
+    ```{.javascript data-prompt=">"}
+    db.getSiblingDB("admin").({
+        "user": "pmm",
+        "pwd": "pmm",
+        "roles": [
+            { "db": "admin", "role": "explainRole" },
+            { "db": "local", "role": "read" },
+            { "db": "admin", "role": "clusterMonitor" }
+        ]
+    })
+    ```
 
-### Set profiling in a `mongo` session
+    If you intend to use PMM's backup management features, also grant these additional permissions: 
+    ```{.javascript data-prompt=">"}
+    db.getSiblingDB("admin").createUser({
+        "user": "pmm",
+        "pwd": "pmm",
+        "roles": [
+            { "db" : "admin", "role": "explainRole" },
+            { "db" : "local", "role": "read" },
+            { "db" : "admin", "role" : "readWrite", "collection": "" },
+            { "db" : "admin", "role" : "backup" },
+            { "db" : "admin", "role" : "clusterMonitor" },
+            { "db" : "admin", "role" : "restore" },
+            { "db" : "admin", "role" : "pbmAnyAction" }
+        ]
+    })      
+    ```
 
-In a `mongo` session, the profiler should be enabled **per** database.
-For example, to enable the profiler in the `testdb`, run this:
 
-```json
-use testdb
-db.setProfilingLevel(2, {slowms: 0})
-```
 
-!!! note alert alert-primary ""
-    If you have already [added a service](#add-service), you should remove it and re-add it after changing the profiling level.
+## Enable MongoDB profiling for Query Analytics (QAN)
 
-## Add service
+To use PMM QAN, you must turn on MongoDB's [profiling feature]. By default, profiling is turned off as it can adversely affect the performance of the database server.
 
-When you have configured your database server, you can add a MongoDB service with the user interface or on the command line.
+Choose one of the following methods to enable profiling:
 
-!!! caution alert alert-warning "Important"
-    To monitor MongoDB sharded clusters, PMM requires access to all cluster components. Make sure to add all config servers, shards, and mongos. Otherwise, PMM will not be able to correctly collect metrics and populate dashboards.
+=== "In MongoDB configuration file (Recommended)"
+    This method ensures your settings persist across server restarts and system reboots. It's the recommended approach for production environments:
+    {.power-number}
+    
+    1. Edit the configuration file (usually `/etc/mongod.conf`).
 
-### With the user interface
+    2. Create or add this to the `operationProfiling` section. ([Read more][MONGODB_CONFIG_OP_PROF].)
 
-To add a service with the UI:
-{.power-number}
+        ```yml
+        operationProfiling:
+          mode: all
+          slowOpThresholdMs: 200
+          rateLimit: 100 # (Only available with Percona Server for MongoDB.)
```

Comment:
Maybe it's better to explain these values: slowOpThershold, mode, rateLimit or give a link to MongoDB documentation. https://docs.percona.com/percona-server-for-mongodb/8.0/rate-limit.html?h=profiling